test/e2e/proxy_test.go

```diff
@@ -68,7 +70,400 @@ func skipIfProxyDisabled(t *testing.T) {
+
+	createAgnhostPod(t, data, hostAgnhost0, node0, true)
+	createAgnhostPod(t, data, hostAnghost1, node1, true)
+	t.Run("Host Network Endpoints", func(t *testing.T) {
+		loadBalancerTestCases(t, data, clusterUrl, localUrl, busybox0, busybox1, busybox0IP, busybox1IP, node0, node1)
+	})
+}
+
+func loadBalancerTestCases(t *testing.T, data *TestData, clusterUrl, localUrl, busybox0, busybox1, busybox0IP, busybox1IP, hostname0, hostname1 string) {
+	t.Run("ExternalTrafficPolicy:Cluster/Client:Node", func(t *testing.T) {
+		testLoadBalancerClusterFromNode(t, data, clusterUrl)
+	})
+	t.Run("ExternalTrafficPolicy:Cluster/Client:Pod", func(t *testing.T) {
+		testLoadBalancerClusterFromPod(t, data, clusterUrl, busybox0, busybox1)
+	})
+	t.Run("ExternalTrafficPolicy:Local/Client:Node", func(t *testing.T) {
+		testLoadBalancerLocalFromNode(t, data, localUrl, hostname0, hostname1)
+	})
+	t.Run("ExternalTrafficPolicy:Local/Client:Pod", func(t *testing.T) {
+		testLoadBalancerLocalFromPod(t, data, localUrl, busybox0, busybox1, busybox0IP, busybox1IP, hostname0, hostname1)
+	})
+}
+
+func createLoadBalancerService(t *testing.T, data *TestData, serviceName string, ingressIPs []string, nodeLoadExternal, isIPv6 bool) {
+	ipProctol := corev1.IPv4Protocol
+	if isIPv6 {
+		ipProctol = corev1.IPv6Protocol
+	}
+	_, err := data.createAgnhostLoadBalancerService(serviceName, true, nodeLoadExternal, ingressIPs, &ipProctol)
+	require.NoError(t, err)
+}
+
+func testLoadBalancerClusterFromNode(t *testing.T, data *TestData, url string) {
+	skipIfKubeProxyEnabled(t, data)
+	errMsg := "Server LoadBalancer whose externalTrafficPolicy is Cluster should be able to be connected from Node"
+	probeFromNode(t, nodeName(0), url, errMsg)
+	probeFromNode(t, nodeName(1), url, errMsg)
+}
+
+func testLoadBalancerClusterFromPod(t *testing.T, data *TestData, url, client0, client1 string) {
+	errMsg := "Server LoadBalancer whose externalTrafficPolicy is Cluster should be able to be connected from Pod"
+	probeFromPod(t, data, client0, url, errMsg)
+	probeFromPod(t, data, client1, url, errMsg)
+}
+
+func testLoadBalancerLocalFromNode(t *testing.T, data *TestData, url, hostname0, hostname1 string) {
+	skipIfKubeProxyEnabled(t, data)
+	errMsg := "Server LoadBalancer whose externalTrafficPolicy is Local should be able to be connected from Node"
+	probeHostnameFromNode(t, nodeName(0), url, hostname0, errMsg)
+	probeHostnameFromNode(t, nodeName(1), url, hostname1, errMsg)
+}
+
+func testLoadBalancerLocalFromPod(t *testing.T, data *TestData, url, client0, client1, client0IP, client1IP, hostname0, hostname1 string) {
+	errMsg := "Server NodePort whose externalTrafficPolicy is Local should be able to be connected from Pod "
+	probeHostnameFromPod(t, data, client0, url, hostname0, errMsg)
+	probeHostnameFromPod(t, data, client1, url, hostname1, errMsg)
+	probeClientIPFromPod(t, data, client0, url, client0IP, errMsg)
+	probeClientIPFromPod(t, data, client1, url, client1IP, errMsg)
```

Comment:
Not only the createXXXPod functions, can we avoid creating any tests/functions that just do same things twice and takes double number of parameters? Even though the functions are only used one place, the code and function signature looks really redundant and not generic.